src/Microsoft.AspNetCore.Authorization/AuthorizationHandler.cs

```diff
@@ -6,38 +6,53 @@
 
 namespace Microsoft.AspNetCore.Authorization
 {
+    /// <summary>
+    /// Base class for authorization handlers that need to be called for a specific requirement type.
+    /// </summary>
+    /// <typeparam name="TRequirement">The type of the requirement to handle.</typeparam>
     public abstract class AuthorizationHandler<TRequirement> : IAuthorizationHandler
-        where TRequirement : IAuthorizationRequirement
+            where TRequirement : IAuthorizationRequirement
     {
-        public void Handle(AuthorizationHandlerContext context)
+        async Task IAuthorizationHandler.HandleAsync(AuthorizationHandlerContext context)
```

Comment:
Then the non-async needs to go, and we only call the async one. Having both is nonsensical and ends up with people overriding the wrong one and wondering why it's never called.